common/src/main/java/com/wynntils/overlays/minimap/MinimapOverlay.java

```diff
@@ -242,54 +246,47 @@ private void renderPois(
             cosRotationRadians = (float) -StrictMath.cos(rotationRadians);
         } else {
             sinRotationRadians = 0f;
-            cosRotationRadians = 0f;
+            cosRotationRadians = 1f;
         }
 
         float currentZoom = 1f / zoomRenderScale;
 
         // Get all MapData features as Pois
-        Stream<? extends Poi> poisToRender = Services.MapData.getFeaturesAsPois();
+        Stream<Pair<MapFeature, ResolvedMapAttributes>> mapFeatures = Services.MapData.getFeatures()
```

Comment:
I've been kind of toying with the idea if we should have a ResolvedMapFeature, which basically would correspond to this pair. I think I got stuck when trying to figure out how this would result in a new hierarchy of ResolvedMapLocation etc, that is distinct from the MapFeature hierarchy...

But maybe we should pursue that idea a bit further.

Reply:
Maybe, although I feel like it might get problematic with generics, and not result in much cleaner code.

Reply:
Let's keep with adhoc pairing like this for now.